kotlinlib/src/mill/kotlinlib/contrib/kover/KoverReportModule.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Some parts of this code are taken from lefou/mill-jacoco. Copyright 2021-Present Tobias Roeser.
+ */
+
+package mill
+package kotlinlib.contrib.kover
+
+import mill.api.Result.Success
+import mill.api.{Loose, PathRef}
+import mill.define.{Discover, ExternalModule}
+import mill.eval.Evaluator
+import mill.kotlinlib.contrib.kover.ReportType.Html
+import mill.kotlinlib.{Dep, DepSyntax, Versions}
+import mill.resolve.{Resolve, SelectMode}
+import mill.scalalib.CoursierModule
+import mill.scalalib.api.CompilationResult
+import mill.util.Jvm
+import os.Path
+
+import java.util.Locale
+
+trait KoverReportBaseModule extends CoursierModule {
+
+  private[kover] val reportName = "kover-report"
+
+  /**
+   * Reads the Kover version from system environment variable `KOVER_VERSION` or defaults to a hardcoded version.
+   */
+  def koverVersion: T[String] = Task.Input {
+    Success[String](T.env.getOrElse("KOVER_VERSION", Versions.koverVersion))
+  }
+
+  def koverCliDep: Target[Agg[Dep]] = T {
+    Agg(ivy"org.jetbrains.kotlinx:kover-cli:${koverVersion()}")
+  }
+
+  /**
+   * Classpath for running Kover.
+   */
+  def koverCliClasspath: T[Loose.Agg[PathRef]] = T {
+    defaultResolver().resolveDeps(koverCliDep())
+  }
+}
+
+/**
+ * Allows the aggregation of coverage reports across multi-module projects.
+ *
+ * Once tests have been run across all modules, this collects reports from
+ * all modules that extend [[KoverModule]]. Simply
+ * define a module that extends [[KoverReportModule]] and
+ * call one of the available "report all" functions.
+ *
+ * For example, define the following `kover` module and use the relevant
+ * reporting option to generate a report:
+ * {{{
+ * object kover extends KoverReportModule
+ * }}}
+ *
+ * - mill __.test                 # run tests for all modules
+ * - mill kover.htmlReportAll     # generates report in html format for all modules
+ * - mill kover.xmlReportAll      # generates report in xml format for all modules
```

Comment:
Having the `generates report in html format for all modules` task belong to a particular module's `.kover` submodule is kind of weird. If we're following the style of `ScalafmtModule` and `PalantirFormatModule`, these aggregate methods would belong to the external module via `kotlinlib.contrib.kover.KoverModule/htmlReportAll` or `kotlinlib.contrib.kover.KoverModule/xmlReportAll`, making it clear they're "global" and not owned by any one particular submodule of the build

Reply:
Here I actually followed the idea of https://github.com/com-lihaoyi/mill/blob/bf81800270ece7adcec07215e1b0bf2735baf107/contrib/scoverage/src/mill/contrib/scoverage/ScoverageReport.scala.

Do you propose to take the approach of `mill-jacoco`: https://github.com/lefou/mill-jacoco/blob/c6d81d370bf780d8b972692fb1f7a7fd335958f6/README.adoc?plain=1#L37? This is essentially the same you proposing I believe.

Reply:
Yeah let's go with the mill-jacoco appraoch, which I believe is what we do for scalafmt/palantirformat. mill/contrib/scoverage is pretty old and crufty, so likely it just pre-dates us figuring out the best way to do these things

Reply:
Done. It will be like `mill.kotlinlib.contrib.kover.KoverModule/htmlReportAll`.